src/netex_idf/offers.rs

```diff
@@ -432,19 +446,99 @@ where
         .collect()
 }
 
-fn parse_passenger_stop_assignment<'a, I>(psa_elements: I) -> HashMap<String, String>
+pub fn get_or_create_stop_point(
+    stop_place_ref: &str,
+    stop_points: &mut CollectionWithId<StopPoint>,
+    monomodal_stopareas: &CollectionWithId<MonomodalStopArea>,
+) -> Result<Idx<StopPoint>> {
+    let stop_place_id = stops::extract_monomodal_stop_place_id(stop_place_ref)?;
+    let stop_point_idx = if let Some(sp_idx) = stop_points.get_idx(&stop_place_id) {
+        sp_idx
+    } else {
+        let monomodal_stoparea = monomodal_stopareas
+            .get(&stop_place_id)
+            .ok_or_else(|| format_err!("Failed to find StopPlace {}", stop_place_ref))?;
+        stop_points.push(StopPoint::from(monomodal_stoparea.to_owned()))?
+    };
+    Ok(stop_point_idx)
+}
+
+fn parse_passenger_stop_assignment<'a, I>(
+    psa_elements: I,
+    stop_points: &mut CollectionWithId<StopPoint>,
+    monomodal_stopareas: &CollectionWithId<MonomodalStopArea>,
+) -> HashMap<String, String>
 where
     I: Iterator<Item = &'a Element>,
 {
     psa_elements
         .filter_map(|psa_element| {
+            let psa_id: String = psa_element.attribute("id")?;
             let scheduled_stop_point_ref: String = psa_element
                 .only_child("ScheduledStopPointRef")?
                 .attribute("ref")?;
-            let quay_ref: String = psa_element
+            let get_quay_ref = |element: &Element| -> Option<String> {
+                element.only_child("QuayRef").and_then(|quay_ref_el| {
+                    quay_ref_el.attribute_with("ref", stops::extract_quay_id)
+                })
+            };
+            let mut get_stop_place_ref = |element: &Element| -> Option<String> {
+                element
+                    .only_child("StopPlaceRef")
+                    .and_then(|stop_place_ref_el| stop_place_ref_el.attribute::<String>("ref"))
```

Comment:
You could replace that line with
```suggestion
                    .and_then(|stop_place_ref_el| stop_place_ref_el.attribute_with::<_, _, String>("ref", stops::extract_monomodal_stop_place_id))
```

And then remove this extraction from line 454.

Reply:
I would lose the raw_id (full StopPlaceRef id) in the warning, however very useful in debug. 
I prefer to stay that way.

Reply:
I would lose the raw_id (full StopPlaceRef id) in the warning, however very useful in debug. I prefer to stay that way.
Edit: no, it's good. With have an other wan just after with the complete PassengerStopAssignment id

Reply:
Done in https://github.com/CanalTP/transit_model/pull/600/commits/9e0f33bcd325eab278221b43f23adfd2e2b8d50c 
with some minor corrections (variable names)

Reply:
Nice cleanup on the variable names.